### PR TITLE
Implement test scenarios for user retrieve without filter parameter

### PR DESCRIPTION
**Pull request by IshikaGodage**
- Implemented the test scenarios for user retrieve without filter. 
- There is no tests for user retrieve from a secondary user store.

### CODE REVIEW

File: product-scenarios/1-manage-users-and-roles/1.2-admin-user-manage-user-accounts/1.2.1-manage-users-with-scim2/src/test/java/org/wso2/identity/scenarios/test/scim2/UserRetrieveWithoutFilterParameterTestCase.java [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+    private CloseableHttpClient client;
+    private SCIM2CommonClient scim2Client;
+    private String userId;
+
+    private static final String SCIM2_USERS_LOCATION = "scim2.users.location";
+    private static final String SCIM2_REQUESTS_LOCATION = "scim2.requests.location";
+    private static JSONParser parser = new JSONParser();
+
+    @Factory(dataProvider = "scim2userConfig")
+    public UserRetrieveWithoutFilterParameterTestCase(JSONObject userJSON_1, JSONObject userJSON_2, JSONObject userJSON_3,
```
Comment (vihanga-liyanage) [inline product-scenarios/1-manage-users-and-roles/1.2-admin-user-manage-user-accounts/1.2.1-manage-users-with-scim2/src/test/java/org/wso2/identity/scenarios/test/scim2/UserRetrieveWithoutFilterParameterTestCase.java]: Follow 120 characters per line rule.

File: product-scenarios/1-manage-users-and-roles/1.2-admin-user-manage-user-accounts/1.2.1-manage-users-with-scim2/src/test/java/org/wso2/identity/scenarios/test/scim2/UserRetrieveWithoutFilterParameterTestCase.java [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+            // create users
+            HttpResponse response = scim2Client.provisionUser(client, userJSON, username, password);
+            assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_CREATED,
+                    "User has not been created successfully");
+            JSONObject returnedUserJSON = getJSONFromResponse(response);
+            userId = returnedUserJSON.get(SCIMConstants.ID_ATTRIBUTE).toString();
+
+            // add user id to userIdList array
+            userIdList.add(userId);
```
Comment (vihanga-liyanage) [inline product-scenarios/1-manage-users-and-roles/1.2-admin-user-manage-user-accounts/1.2.1-manage-users-with-scim2/src/test/java/org/wso2/identity/scenarios/test/scim2/UserRetrieveWithoutFilterParameterTestCase.java]: Shouldn't we add the userId to the list after checking the assertions?

File: product-scenarios/1-manage-users-and-roles/1.2-admin-user-manage-user-accounts/1.2.1-manage-users-with-scim2/src/test/java/org/wso2/identity/scenarios/test/scim2/UserRetrieveWithoutFilterParameterTestCase.java [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+    @AfterClass(alwaysRun = true)
+    public void testClear() throws Exception {
+
+        for (String userId : userIdList) {
+
+            // delete created users from database
+            HttpResponse deleteResponse = scim2Client.deleteUser(client, userId, ADMIN_USERNAME, ADMIN_PASSWORD);
+
+            // validate user deletion
+            assertEquals(deleteResponse.getStatusLine().getStatusCode(), HttpStatus.SC_NO_CONTENT, "Failed to delete the user");
```
Comment (vihanga-liyanage) [inline product-scenarios/1-manage-users-and-roles/1.2-admin-user-manage-user-accounts/1.2.1-manage-users-with-scim2/src/test/java/org/wso2/identity/scenarios/test/scim2/UserRetrieveWithoutFilterParameterTestCase.java]: Looks like this line has more than 120 characters.

File: product-scenarios/1-manage-users-and-roles/1.2-admin-user-manage-user-accounts/1.2.1-manage-users-with-scim2/src/test/java/org/wso2/identity/scenarios/test/scim2/UserRetrieveWithoutFilterParameterTestCase.java [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+            JSONObject resultsJson = (JSONObject)requestJson.get(RESULTS_ATTRIBUTE);
+
+            String startIndex = paramsJson.get(SCIM2_QUERY_PARAM_START_INDEX_ATTRIBUTE).toString();
+            String count = paramsJson.get(SCIM2_QUERY_PARAM_COUNT_ATTRIBUTE).toString();
+            String domain = null;
+            if (paramsJson.get(SCIM2_QUERY_PARAM_DOMAIN_ATTRIBUTE) != null) {
+                domain = paramsJson.get(SCIM2_QUERY_PARAM_DOMAIN_ATTRIBUTE).toString();
+            }
+
+            HttpResponse response = scim2Client.getUser(client, startIndex, count, domain, ADMIN_USERNAME, ADMIN_PASSWORD);
```
Comment (vihanga-liyanage) [inline product-scenarios/1-manage-users-and-roles/1.2-admin-user-manage-user-accounts/1.2.1-manage-users-with-scim2/src/test/java/org/wso2/identity/scenarios/test/scim2/UserRetrieveWithoutFilterParameterTestCase.java]: Looks like this line has more than 120 characters.

File: product-scenarios/1-manage-users-and-roles/1.2-admin-user-manage-user-accounts/1.2.1-manage-users-with-scim2/src/test/java/org/wso2/identity/scenarios/test/scim2/UserRetrieveWithoutFilterParameterTestCase.java [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+
+    /**
+     * Validate response.
+     *
+     * @param paramsJson      Query Parameters.
+     * @param resultsJSON     Expected Results.
+     * @param applicationJSON Results of Response.
+     * @throws Exception      Exception.
+     */
+    private void validateResponse(JSONObject paramsJson, JSONObject resultsJSON, JSONObject applicationJSON) throws IOException {
```
Comment (vihanga-liyanage) [inline product-scenarios/1-manage-users-and-roles/1.2-admin-user-manage-user-accounts/1.2.1-manage-users-with-scim2/src/test/java/org/wso2/identity/scenarios/test/scim2/UserRetrieveWithoutFilterParameterTestCase.java]: Please limit the character count to 120 per line.

File: product-scenarios/1-manage-users-and-roles/1.2-admin-user-manage-user-accounts/1.2.1-manage-users-with-scim2/src/test/java/org/wso2/identity/scenarios/test/scim2/UserRetrieveWithoutFilterParameterTestCase.java [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+                        "UserRetrieveWithoutFilterParameterTestCase. Requested users count: "
+                        + paramsJson.get(SCIM2_QUERY_PARAM_COUNT_ATTRIBUTE).toString() + ", Received user count: "
+                        + applicationJSON.get(SCIM2_RESPONCE_TOTAL_RESULTS_ATTRIBUTE).toString() );
+
+        // validate users startIndex
+        assertEquals(applicationJSON.get(SCIM2_RESPONCE_START_INDEX_ATTRIBUTE).toString(),
+                paramsJson.get(SCIM2_QUERY_PARAM_START_INDEX_ATTRIBUTE).toString(), "StartIndex is invalid in " +
+                        "UserRetrieveWithoutFilterParameterTestCase. Requested startIndex: "
+                        + paramsJson.get(SCIM2_QUERY_PARAM_START_INDEX_ATTRIBUTE).toString()
+                        + ", Received startIndex: " + applicationJSON.get(SCIM2_RESPONCE_START_INDEX_ATTRIBUTE).toString() );
```
Comment (vihanga-liyanage) [inline product-scenarios/1-manage-users-and-roles/1.2-admin-user-manage-user-accounts/1.2.1-manage-users-with-scim2/src/test/java/org/wso2/identity/scenarios/test/scim2/UserRetrieveWithoutFilterParameterTestCase.java]: Please limit the character count to 120 per line.

File: product-scenarios/1-manage-users-and-roles/1.2-admin-user-manage-user-accounts/1.2.1-manage-users-with-scim2/src/test/java/org/wso2/identity/scenarios/test/scim2/UserRetrieveWithoutFilterParameterTestCase.java [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+            JSONObject resourceJson = (JSONObject)resource.get(i);
+            // validate id
+            String userId = resourceJson.get(SCIM2_RESPONCE_USER_ID_ATTRIBUTE).toString();
+            assertNotNull(userId, "SCIM2 user id not available in the response.");
+
+            // validate user name
+            JSONObject user = (JSONObject)userNamesJson.get(i);
+            assertEquals(resourceJson.get(SCIM2_RESPONCE_USER_NAME_ATTRIBUTE).toString(), user.get(NAME_ATTRIBUTE).toString(),
+                    "Received user name is invalid in UserRetrieveWithoutFilterParameterTestCase. " +
+                            "Requested user name: " + user.get(NAME_ATTRIBUTE).toString() + ", Received user name: "
```
Comment (vihanga-liyanage) [inline product-scenarios/1-manage-users-and-roles/1.2-admin-user-manage-user-accounts/1.2.1-manage-users-with-scim2/src/test/java/org/wso2/identity/scenarios/test/scim2/UserRetrieveWithoutFilterParameterTestCase.java]: Please limit the character count to 120 per line.

File: product-scenarios/1-manage-users-and-roles/1.2-admin-user-manage-user-accounts/1.2.1-manage-users-with-scim2/src/test/java/org/wso2/identity/scenarios/test/scim2/SCIMConstants.java [hunk 8]
```diff
@@ -153,4 +152,20 @@
    public static final String USER_SCHEMA = "urn:ietf:params:scim:schemas:core:2.0:User";
    public static final String ERROR_SCHEMA = "urn:ietf:params:scim:api:messages:2.0:Error";
    public static final String ENTERPRISE_SCHEMA = "urn:scim:schemas:extension:enterprise:1.0";
+
+   public static final String SCIM2_RESPONCE_TOTAL_RESULTS_ATTRIBUTE = "totalResults";
```
Comment (Krzysztof1985) [inline product-scenarios/1-manage-users-and-roles/1.2-admin-user-manage-user-accounts/1.2.1-manage-users-with-scim2/src/test/java/org/wso2/identity/scenarios/test/scim2/SCIMConstants.java]: Small typo ;) I think it should be 'RESPONSE' 

Kind Regards
Krzysztof

**Comment by IshikaGodage** [comment]
Since These test cases are no longer valid, closed the PR.

**Pull request closed without merging**